port/zos390/omrvmem.c

```diff
@@ -1296,30 +1300,37 @@ omrvmem_default_large_page_size_ex(struct OMRPortLibrary *portLibrary, uintptr_t
 	}
 #endif /* OMR_ENV_DATA64 */
 
-	while (0 != PPG_vmem_pageSize[index]) {
-		if (0 != (OMRPORT_VMEM_MEMORY_MODE_EXECUTE & mode)) {
-			if ((ONE_M == PPG_vmem_pageSize[index]) &&
-				(0 != (OMRPORT_VMEM_PAGE_FLAG_PAGEABLE & PPG_vmem_pageFlags[index]))
-			) {
-				break;
+	for (int i = 0; ((0 != PPG_vmem_pageSize[i]) && (ONE_M >= PPG_vmem_pageSize[i])); i++) {
+		if (ONE_M == PPG_vmem_pageSize[i]) {
+			if (0 != (OMRPORT_VMEM_PAGE_FLAG_PAGEABLE & PPG_vmem_pageFlags[i])) {
+				indexPageable = i;
 			}
 #if defined(OMR_ENV_DATA64)
-		} else {
-			if ((ONE_M == PPG_vmem_pageSize[index]) &&
-				(0 != (OMRPORT_VMEM_PAGE_FLAG_FIXED & PPG_vmem_pageFlags[index]))
-			) {
-				break;
+			else if (0 != (OMRPORT_VMEM_PAGE_FLAG_FIXED & PPG_vmem_pageFlags[i])) {
+				index = i;
 			}
-#endif
+#endif /* OMR_ENV_DATA64 */
 		}
-		index++;
 	}
-	if (NULL != pageSize) {
-		*pageSize = PPG_vmem_pageSize[index];
+
+	/* 
+	 * When mode is OMRPORT_VMEM_MEMORY_EXECUTE we will only consider pagable large pages. Return pageSize of 0 if 1M pageable is not configured on the system.
+	 * When flag is OMRPORT_VMEM_PAGE_FLAG_PAGEABLE_PREFERABLE, we prefer pageable 1M large pages unless they're not provisioned.
+	 */
+	if ((0 != (OMRPORT_VMEM_MEMORY_MODE_EXECUTE & mode)) ||
+	   (TRUE == pageablePreferableRequested && (-1 != indexPageable))) {
```

Comment:
cosmetic: there is no need to compare BOOLEAN with TRUE, so you can use:
```
	if ((0 != (OMRPORT_VMEM_MEMORY_MODE_EXECUTE & mode)) ||
 	   (pageablePreferableRequested && (-1 != indexPageable))) {
```

Reply:
I was trying to safe, is what you're suggesting known to be safe? Everywhere else in this file they compare to TRUE/FALSE.

Reply:
see that construction TRUE == is used in a few places already, so I guess it is good as is. Ignore this comment